ethylene-core/src/main/java/com/github/steanky/ethylene/core/mapper/BasicTypeResolver.java

```diff
@@ -0,0 +1,106 @@
+package com.github.steanky.ethylene.core.mapper;
+
+import org.apache.commons.lang3.ClassUtils;
+import org.apache.commons.lang3.reflect.TypeUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.lang.ref.Reference;
+import java.lang.ref.WeakReference;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.WeakHashMap;
+import java.util.function.Supplier;
+
+public class BasicTypeResolver implements TypeResolver {
```

Comment:
I think this could benefit from either a static constructor-like (preferably) method or an actual constructor that registers some frequently used java collections and maps